src/ports/database.ts

```diff
@@ -0,0 +1,53 @@
+import type SystemRepository from './system/interface';
+import type UserRepository from './user/interface';
+import type { Identity, Undefined, Lang, String } from './types';
+import type { User } from './user/model';
+import type { System } from './system/model';
+
```

Comment:
question about the folder structure.
what does `ports` mean in this context?
maybe everything database related should be moved into a single folder to not "pollute" the folder structure